ipv8/attestation/trustchain/community.py

```diff
@@ -51,32 +54,42 @@ class TrustChainCommunity(Community):
 
     DB_CLASS = TrustChainDB
     DB_NAME = 'trustchain'
-    BROADCAST_FANOUT = 10
     version = '\x02'
 
     def __init__(self, *args, **kwargs):
         working_directory = kwargs.pop('working_directory', '')
         db_name = kwargs.pop('db_name', self.DB_NAME)
+        self.settings = kwargs.pop('settings', TrustChainSettings())
         super(TrustChainCommunity, self).__init__(*args, **kwargs)
         self.request_cache = RequestCache()
         self.logger = logging.getLogger(self.__class__.__name__)
         self.persistence = self.DB_CLASS(working_directory, db_name)
         self.relayed_broadcasts = []
         self.logger.debug("The trustchain community started with Public Key: %s",
                           hexlify(self.my_peer.public_key.key_to_bin()))
-        self.broadcast_block = True  # Whether we broadcast a full block after constructing it
         self.shutting_down = False
         self.listeners_map = {}  # Map of block_type -> [callbacks]
+        self.db_cleanup_lc = self.register_task("db_cleanup", LoopingCall(self.do_db_cleanup))
```

Comment:
As an alternative, we could trigger a database cleanup every X blocks but a large arrival rate of new blocks will trigger the cleanup more frequently, impacting performance. This avoids the looping call though.

Reply:
I was thinking of offering something like we have for peer discovery (`DiscoveryStrategy`). If we keep all of the timing stuff on a higher level, the communities also don't have to do their own call management (`register_task`). This would make it harder to introduce errors.